### PR TITLE
Synchronize tark started notification with task execution submitted t…

### PR DESCRIPTION
**Pull request by jrochas**
…o TaskLauncher

- the notification and persistence of the task started status was done in parallel with the submission of the task to the TaskLauncher. This small gap led to inconsistencies, especially in the case of a crash of the scheduler within this gap. This commit introduces a small synchronization to be sure to notify the task started status once the doTask request has been received by the TaskLauncher.
- the existing TaskRecoveryWhenNodesAreReservedInBatchTest from commit 018b485c014c48cf8e4edda66931452d10c9aeaa exhibits this particular case.

### CODE REVIEW

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java [hunk 1]
```diff
@@ -653,23 +658,27 @@ protected boolean createExecution(NodeSet nodeSet, Node node, InternalJob job, I
...
-                                                                       terminateNotification,
-                                                                       corePrivateKey,
-                                                                       terminateNotificationNodeURL),
-                                                 DOTASK_ACTION_TIMEOUT,
-                                                 TimeUnit.MILLISECONDS);
-
-                    finalizeStarting(job, task, node, launcher);
-
-                    return true;
+                    try {
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java]: MAJOR Extract this nested try block into a separate method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java [hunk 2]
```diff
@@ -692,6 +693,17 @@ protected boolean createExecution(NodeSet nodeSet, Node node, InternalJob job, I
...
     }
 
+    private void waitForTaskToBeStarted(Future<Void> taskExecutionSubmittedFuture) {
+        try {
+            // before signaling that the task is started, we need
+            // to make sure the task is correctly submitted to the
+            // task launcher
+            taskExecutionSubmittedFuture.get(DOTASK_ACTION_TIMEOUT, TimeUnit.MILLISECONDS);
+        } catch (Exception e) {
+            logger.warn("Could not wait for the task to be started on TaskLauncher.", e);
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java]: I prefer "Error while waiting for the task to be started.", the exception should inform if it's a timeout, etc.

Comment (jrochas) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java]: ok, I'll change that into multiple catch then.

**Pull request merged**